crates/deno-subsystem/deno-model-builder/src/module_skeleton_generator.rs

```diff
@@ -436,3 +494,203 @@ fn typescript_base_type(exo_type_name: &str) -> String {
         t => t.to_string(),
     }
 }
+
+mod tests {
+    use super::*;
+
+    struct MockType {
+        name: String,
+        fields: Vec<(String, String)>,
+    }
+
+    impl Type for MockType {
+        fn name(&self) -> &str {
+            &self.name
+        }
+
+        fn fields(&self) -> Vec<(&str, &dyn TypeScriptType)> {
+            self.fields
+                .iter()
+                .map(|(name, typ)| (name.as_str(), typ as &dyn TypeScriptType))
+                .collect()
+        }
+    }
+
+    impl TypeScriptType for String {
+        fn typescript_type(&self) -> String {
+            self.clone()
+        }
+    }
+
+    #[test]
+    fn test_generate_type_skeleton() {
+        use tempfile::tempfile;
+        use std::io::Seek;
+        use std::io::Read;
+
+        let mock_type = MockType {
+            name: "TestType".to_string(),
+            fields: vec![
+                ("field1".to_string(), "string".to_string()),
+                ("field2".to_string(), "number".to_string()),
```

Comment:
If you change the type from `number` to `Int` or `Float`, that will be accurately reflect user code.